scalafmt-core/shared/src/main/scala/org/scalafmt/rewrite/RedundantParens.scala

```diff
@@ -60,13 +60,24 @@ object RedundantParens extends Rewrite with FormatTokensRewrite.RuleFactory {
         case p: Type.FunctionType => iff(p.res)
         case p: Type.PolyFunction => iff(p.tpe)
         case _: Term.Return | _: Term.Throw | _: Term.QuotedMacroExpr |
-            _: Term.SplicedMacroExpr =>
+            _: Term.SplicedMacroExpr | _: Term.Block =>
           Some(true)
         case _ => None
       }
     }
   }
 
+  object IsInBraces {
+    def unapply(t: Tree): Option[Boolean] = t match {
+      case _: Term.Block | _: Term.PartialFunction =>
```

Comment:
Will this work with optional braces?

Reply:
can you have parens around optional braces?

Reply:
Ach, it cant's from what I know, so this is just the situation with `({ ... })` ? In that case it's fine